### PR TITLE
use localref for imagestreams

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@coreydaley ptal
@gabemontero fyi this also pulls in your jenkins template changes, we can see which one merges first.

**Comment by bparees** [comment]
/hold
(for passing extended tests)

**Comment by coreydaley** [comment]
@bparees I'm guessing that you did try loading these json files into OpenShift to validate that they are valid, if so, then /lgtm

**Comment by bparees** [comment]
> @bparees I'm guessing that you did try loading these json files into OpenShift to validate that they are valid, if so, then /lgtm

that is what our tests do.

**Comment by coreydaley** [comment]
Looks like the mysql server failed to start, got stuck in ContainerCreating

/retest

**Comment by bparees** [comment]
@coreydaley can you give this a final review/sign off?

File: test/cmd/newapp.sh [hunk 1]
```diff
@@ -360,8 +360,8 @@ os::cmd::expect_failure_and_text 'oc new-app --search mysql --param=FOO=BAR' "ca
 os::cmd::expect_failure_and_not_text 'oc new-app --template foo' 'index out of range'
 
 # set context-dir
-os::cmd::expect_success_and_text 'oc new-app https://github.com/openshift/sti-ruby.git --context-dir="2.3/test/puma-test-app" -o yaml' 'contextDir: 2.3/test/puma-test-app'
-os::cmd::expect_success_and_text 'oc new-app ruby~https://github.com/openshift/sti-ruby.git --context-dir="2.3/test/puma-test-app" -o yaml' 'contextDir: 2.3/test/puma-test-app'
+os::cmd::expect_success_and_text 'oc new-app https://github.com/sclorg/s2i-ruby-container.git --context-dir="2.4/test/puma-test-app" -o yaml' 'contextDir: 2.4/test/puma-test-app'
+os::cmd::expect_success_and_text 'oc new-app ruby~https://github.com/s2i-ruby-container.git --context-dir="2.4/test/puma-test-app" -o yaml' 'contextDir: 2.4/test/puma-test-app'
```
Comment (coreydaley) [inline test/cmd/newapp.sh]: I think you are missing the org here

Comment (bparees) [reply test/cmd/newapp.sh]: interestingly that doesn't actually matter for this test, but thanks i'll fix it :)

Comment (bparees) [reply test/cmd/newapp.sh]: updated.

**Pull request merged**